### PR TITLE
Settings: Remove Markdown Settings

### LINKED ISSUE #6669: Remove markdown from JP?
WordPress.com is [removing the markdown settings](https://github.com/Automattic/wp-calypso/pull/12143). Should we follow suit? I am all for removing these settings if there are no major objections.

To clarify, we would enable markdown on both posts and comments and simply remove the settings for them.

CC @rickybanister @jeffgolenski @georgestephanis @samhotchkiss

### PR DESCRIPTION
**Pull request by ockham**
Fixes #6669. For rationale, see discussion there.

### Testing instructions:

On the Settings page:
* In the 'Writing' tab, verify that the 'Write posts or pages in plain-text Markdown syntax' option is gone.
* In the 'Discussion' tab, verify that the 'Enable Markdown use for comments' toggle is gone.
* Verify that modifying and saving other settings still works.

#### Before:

image

image

#### After:

image

image

#### Proposed changelog entry for your changes:

* Removed Markdown Settings

### CODE REVIEW

**Comment by ebinnion** [comment]
There are a few more instances of markdown that we should be able to remove:

- `_inc/client/discsussion/index.jsx`
- `_inc/client/components/navigation-settings/index.jsx`
- `_inc/client/writing/index.jsx`

File: _inc/client/discussion/comments.jsx [hunk 1]
```diff
@@ -19,27 +19,12 @@ import SettingsCard from 'components/settings-card';
...
-	};
-
 	render() {
 		const foundComments = this.props.isModuleFound( 'comments' ),
 			foundGravatar = this.props.isModuleFound( 'gravatar-hovercards' ),
-			foundMarkdown = this.props.isModuleFound( 'markdown' ),
 			foundCommentLikes = this.props.isModuleFound( 'comment-likes' );
 
-		if ( ! foundComments && ! foundGravatar && ! foundMarkdown && ! foundCommentLikes ) {
+		if ( ! foundComments && ! foundGravatar && && ! foundCommentLikes ) {
```
Comment (ebinnion) [inline _inc/client/discussion/comments.jsx]: This is causing an error on `yarn build` due to the double `&&`.

Comment (ockham) [reply _inc/client/discussion/comments.jsx]: D'oh, amended the commit but forgot to force-push.

**Comment by ockham** [comment]
> There are a few more instances of markdown that we should be able to remove:
> 
> - `_inc/client/discsussion/index.jsx`
> - `_inc/client/components/navigation-settings/index.jsx`
> - `_inc/client/writing/index.jsx`

Oops, I should've grepped myself 🙄 

Done now!

**Comment by MichaelArestad** [comment]
@ockham Is this defaulting to on for folks like on dotcom?

@beaulebens thoughts/concerns?

**Comment by samhotchkiss** [comment]
I have a couple of concerns about this:

1. This could cause interference if the user is running another plugin which parses markdown.  We do our best to auto-detect this, but won't always be able to.
2. A user could be intentionally using underscores or asterisks in a way that would be parsed as markdown, and they have no way of knowing what's happening or turning it off.

I would propose that, if we want to ship this to users (which I'd be all about, as I love markdown, personally), we do one or both of two things:

1. Add a JITM which pops up once they've saved a post which our markdown parser matched content on, telling them that it looks like they wrote some markdown, we're handling it automatically, and provide them with a way to disable it.
2. When we do detect markdown on their site, set some sort of flag which, when set, gives them the option to disable markdown.

That said, my #2 concern may be a non-issue, if it hasn't been a problem for WP.com since introducing this change there last year.  I'd recommend checking in with wpcom HE's to see if this is something that comes up with any regularity.

**Comment by MichaelArestad** [comment]
> This could cause interference if the user is running another plugin which parses markdown. We do our best to auto-detect this, but won't always be able to.

@samhotchkiss I have that concern, too, but this would only effect sites with new JP installs. It's defaulting to on rather than being forced on. I suspect we won't hear about many issues with this.

@gravityrail Do have a way to see how many JP sites have some sort of markdown plugin installed? Of those, how many are incompatible (that we're aware of)?

> A user could be intentionally using underscores or asterisks in a way that would be parsed as markdown, and they have no way of knowing what's happening or turning it off.

Several characters already are parsed as markdown (sort of) in the core editor. We also defaulted it to on for WordPress.com quite some time ago and I have not heard anything around folks complaining about it.

In a Gutenberg world, I'm hoping we scope markdown to just markdown blocks (if we eventually build one) and the Classic editor blocks.

**Comment by samhotchkiss** [comment]
Hey @MichaelArestad -- @eliorivero just reminded me that we have some Jetpack features which can only be seen if you search for them.  Would you be okay with making Markdown one of these?  That way users can find/deactivate if they run into issues.

**Comment by MichaelArestad** [comment]
@samhotchkiss That is indeed how this should work, though I don't think they allow for deactivation as much as just reactivation. (at least currently)

**Comment by samhotchkiss** [comment]
So, if I have a Jetpack site currently, once this upgrade is made, how do I activate MD for the first time?

**Comment by MichaelArestad** [comment]
@samhotchkiss default on. On dotcom it's default on. Perhaps with the advent of Gutenberg, we should default it off instead?

**Comment by samhotchkiss** [comment]
So, @MichaelArestad, does that mean we're going to go back and turn it on for any Jetpack site when they upgrade to JP 6.0?

Have we definitively decided that we're not going to parse Markdown in normal Gutenblocks?

**Comment by MichaelArestad** [comment]
> does that mean we're going to go back and turn it on for any Jetpack site when they upgrade to JP 6.0?

@samhotchkiss No. We are not changing it's state for sites already active. Only new setups. We've done this with some previous modules.

> Have we definitively decided that we're not going to parse Markdown in normal Gutenblocks?

There have been no decisions. I don't think anyone has looked into feasibility of it either way.

**Comment by MichaelArestad** [comment]
@samhotchkiss sounds like Gutenberg blocks already parse markdown so we won't need to have it all if that remains true on launch.

**Comment by samhotchkiss** [comment]
> No. We are not changing it's state for sites already active. Only new setups. We've done this with some previous modules.

@MichaelArestad so, how would an older site activate MD?  will they still see the option?  (I remember there were a couple other modules we did that for)

**Comment by MichaelArestad** [comment]
@samhotchkiss it would show up in search like JSON API or something that was disabled but no longer has settings. There is an Enable button on those previously disabled.

**Comment by samhotchkiss** [comment]
@MichaelArestad -- if you think that's best, I'm good with it.  Personally, I'd rather it show on old sites until they enable it.

**Comment by oskosk** [comment]
I've just rebased this so it can get rebuilt by the Beta server

**Comment by oskosk** [comment]
I rebased this one as it showed conflicts now

**Comment by ockham** [comment]
Closing for now. Looks like this wasn't exactly a low-hanging fruit for Hack Week after all.

**Pull request closed without merging**